MetaGenerator/DotnetToolInstaller.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Messerli.MetaGeneratorAbstractions;
+using Messerli.ToolLoaderAbstractions;
+
+namespace Messerli.MetaGenerator
+{
+    public class DotnetToolInstaller : IDotnetToolInstaller
+    {
+        private const string Dotnet = "dotnet";
+
+        private readonly ITools _tools;
+
+        public DotnetToolInstaller(ITools tools)
+        {
+            _tools = tools;
+            _tools.RegisterTool(Dotnet, "dotnet.exe");
+            _tools.VerifyTools();
+        }
+
+        public void InstallTool(string path, string toolName, string? version)
+        {
+            const string dotnetToolsFileName = "dotnet-tools.json";
+            var dotnet = _tools.GetTool("dotnet.exe");
+
+            var hasDotnetToolManifest = Directory.GetFiles(path, dotnetToolsFileName, SearchOption.AllDirectories)
+                .Any();
+
+            if (!hasDotnetToolManifest)
+            {
+                dotnet.Execute(new[] { "new", "tool-manifest" }, path);
+            }
+
+            dotnet.Execute(GetToolInstallArguments(toolName, version), path);
```

Comment:
Move the execute calls into own methods. There's no benefit to sharing the `dotnet` object, as it will be used exactly once anyways.

Reply:
Addressed in: [Refactor DotnetToolInstaller](https://github.com/messerli-informatik-ag/meta-generator/pull/24/commits/0bbe55b4481be7d90539bb1b81a03793c0b95cb9)